strawberry/annotation.py

```diff
@@ -81,18 +85,29 @@ def from_annotation(
             return StrawberryAnnotation(annotation, namespace=namespace)
         return annotation
 
-    def resolve(self) -> Union[StrawberryType, type]:
+    def evaluate(self) -> type:
+        """Return evaluated annotation using `strawberry.util.typing.eval_type`."""
+        evaled_type = self.__eval_cache__
+        if evaled_type:
+            return evaled_type
+
         annotation = self.annotation
         if isinstance(annotation, str):
             annotation = ForwardRef(annotation)
 
-        args = []
-
         evaled_type = eval_type(annotation, self.namespace, None)
 
+        self.__eval_cache__ = evaled_type
+        return evaled_type
```

Comment:
couldn't we use cached property here?

Reply:
We'd have to remove the `__slots__` declaration above which is something I've
been wanting to introduce to Strawberry classes for a while. Especially `Info`
as it would reduce memory usage

Reply:
We can use cached property if you want to 😊

Reply:
oh, I didn't realise __slots__ don't make cachedproperty work! that's fine then